crates/rwasm/executor/src/events/byte.rs

```diff
@@ -27,9 +27,11 @@ pub struct ByteLookupEvent {
     /// The fourth operand.
     pub c: u8,
 }
+use core::any::Any;
 
 /// A type that can record byte lookup events.
-pub trait ByteRecord {
+pub trait ByteRecord: Any {
+    fn as_any(&self) -> &dyn Any;
```

Comment:
why not make this is_dummy as boolean function and only make emptybyterecord return false.